contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -2540,6 +2540,143 @@ CREATE OR REPLACE FUNCTION sys.json_query(json_string text, path text default '$
+
+CREATE OR REPLACE FUNCTION sys.set_json_modify_flags(path_json TEXT)
+RETURNS json_modify_flags
+AS
+$BODY$
+DECLARE
+    result_mode_flags json_modify_flags;
+    word_count INTEGER;
+    create_if_missing BOOL:='TRUE';
+    mode TEXT;
+    modifier TEXT;
+    append_modifier BOOL:='FALSE';
+BEGIN
+    
+    word_count = LENGTH(path_json) - LENGTH(REPLACE(path_json, ' ', '')) + 1; --To calculate the number of words in the path_json
+    
+    -- This if else block is added to set the create_if_missing and append_modifier flags
+    IF word_count=1 THEN --The path_json has only 1 word
+        create_if_missing='TRUE';
+        append_modifier='FALSE';
+    ELSIF word_count=2 THEN --The path_json has 2 words
+        mode = SPLIT_PART(path_json COLLATE sql_latin1_general_cp1_cs_as, ' ',1);
+        IF mode='append' THEN
+            append_modifier:='TRUE';
+        ELSIF mode='strict' THEN
+            create_if_missing:='FALSE';
+        END IF;
+    ELSIF word_count=3 THEN --The path_json has 3 words
+        modifier = SPLIT_PART(path_json COLLATE sql_latin1_general_cp1_cs_as, ' ',1);
+        mode = SPLIT_PART(path_json COLLATE sql_latin1_general_cp1_cs_as, ' ',2);
+        IF modifier='append' THEN
+            append_modifier:='TRUE';
+        END IF;
+        IF mode='strict' THEN
+            create_if_missing:='FALSE';
+        END IF;
+    END IF;
+    SELECT create_if_missing,append_modifier INTO result_mode_flags.create_if_missing, result_mode_flags.append_modifier;
+    RETURN result_mode_flags;
+END
+$BODY$
+LANGUAGE plpgsql;
+
+
+--JSON_MODIFY
+CREATE OR REPLACE FUNCTION sys.json_modify(in json_string NVARCHAR,in path_json TEXT, in new_value TEXT)
+RETURNS sys.NVARCHAR
+AS
+$BODY$
+DECLARE
+    json_path TEXT;
+    mode_flags json_modify_flags;
+    new_jsonb_path TEXT;
+    len_array INTEGER;
+    type_key_value TEXT;
+    key_exist BOOL;
+    key_value NVARCHAR;
+BEGIN
+    json_path = regexp_replace(path_json, '^.* ', '','ig'); -- To select the last word of the json_path string. \\\\\\Instead of this one can use split_part if regexp is slow
+    new_jsonb_path = sys.convert_jsonpath_to_jsonb_path(json_path); --calling function to convert the path format
+    key_exist = jsonb_path_exists(json_string::JSONB,json_path::jsonpath); -- TO check if key exist in the given path
+
+    mode_flags = sys.set_json_modify_flags(path_json); --Calling a function to set the flags
+    
+    
+    --This if else block is to call the jsonb_set function based on the create_if_missing and append_modifier flags
+    
+    IF mode_flags.append_modifier THEN --When append modifier is present
+        IF key_exist THEN
+            key_value = jsonb_path_query_first(json_string::JSONB,json_path::jsonpath); -- To get the value of the key
+            type_key_value = jsonb_typeof(key_value::JSONB);
+            IF type_key_value='array' THEN
+                len_array = jsonb_array_length(key_value::JSONB);
```

Comment:
small nitpick: casting is expensive, so try to cast once and save the value in a variable for reuse instead of casting types multiple times

Reply:
Have made changes in the code to reduce the number of cast done.